examples/metrics/stateful.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+This module serves as an example for a simple application using metrics
+Examples show how to recording affects the collection of metrics to be exported
+"""
+import time
+
+from opentelemetry import metrics
+from opentelemetry.sdk.metrics import Counter, Meter
+from opentelemetry.sdk.metrics.export import ConsoleMetricsExporter
+from opentelemetry.sdk.metrics.export.batcher import UngroupedBatcher
+from opentelemetry.sdk.metrics.export.controller import PushController
+
+# Batcher used to collect all created metrics from meter ready for exporting
+# Pass in true/false to indicate whether the batcher is stateful. True
+# indicates the batcher computes checkpoints from over the process lifetime.
+# False indicates the batcher computes checkpoints which describe the updates
+# of a single collection period (deltas)
+batcher = UngroupedBatcher(True)
+# If a batcher is not provded, a default batcher is used
+# Meter is responsible for creating and recording metrics
+meter = Meter(batcher)
+metrics.set_preferred_meter_implementation(meter)
+# exporter to export metrics to the console
+exporter = ConsoleMetricsExporter()
+# controller collects metrics created from meter and exports it via the
+# exporter every interval
+controller = PushController(meter, exporter, 5)
+
+counter = meter.create_metric(
+    "available memory",
+    "available memory",
+    "bytes",
+    int,
+    Counter,
+    ("environment",),
+)
+
+counter2 = meter.create_metric(
```

Comment:
Why is the "stateful" example different from the "non-stateful" one in ways other than batcher state?

If the goal is to compare stateful and non-stateful batchers, what do you think about combining these examples to show only that?

Reply:
Combining the two examples will require metrics to load a different implementation of `Meter` since a `Meter` instance is initialized with a certain `Batcher`. The second example can be run after the first sequentially but the example file would get huge and quite confusing to read. I'd prefer to separate the two examples.